contentcuration/contentcuration/management/commands/set_orm_based_has_captions.py

```diff
@@ -0,0 +1,50 @@
+import logging as logmodule
+import time
+
+from django.core.management.base import BaseCommand
+from django.db.models import Exists
+from django.db.models import OuterRef
+from le_utils.constants import content_kinds
+from le_utils.constants import format_presets
+from le_utils.constants.labels import accessibility_categories
+
+from contentcuration.models import ContentNode
+from contentcuration.models import File
+
+logmodule.basicConfig(level=logmodule.INFO)
+logging = logmodule.getLogger('command')
+
+
+CHUNKSIZE = 10000
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        start = time.time()
+
+        logging.info("Setting 'has captions' for video kinds")
+
+        # Only try to update video nodes which have not had any accessibility labels set on them
+        # this will allow this management command to be rerun and resume from where it left off
+        # and also prevent stomping previous edits to the accessibility_labels field.
+        updateable_nodes = ContentNode.objects.annotate(
+            has_captions=Exists(File.objects.filter(contentnode=OuterRef("id"), language=OuterRef("language"), preset_id=format_presets.VIDEO_SUBTITLE))
+        ).filter(kind=content_kinds.VIDEO, accessibility_labels__isnull=True, has_captions=True)
+
+        updateable_node_slice = list(updateable_nodes.values_list("id", flat=True)[0:CHUNKSIZE])
```

Comment:
It probably could be left as a subquery - but just because I was using it for an existence check, I had materialized it to a list first, and so I thought I would reduce the complexity of the update query by just passing in the list of ids.

Reply:
Changed the `while` check to an existence check on the whole query, and removed the coersion to a list.